### PR TITLE
DM-32412: Add storageClass and formatter for cell-based coadds

### PR DESCRIPTION
**Pull request by arunkannawadi**
## Checklist

- [x] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/daf/butler/configs/storageClasses.yaml [hunk 1]
```diff
@@ -282,3 +282,5 @@ storageClasses:
     pytype: lsst.atmospec.spectraction.Spectraction
   ScarletModelData:
     pytype: lsst.meas.extensions.scarlet.ScarletModelData
+  MultipleCellCoadd:
+    pytytpe: lsst.cell_coadds.MultipleCellCoadd
```
Comment (timj) [inline python/lsst/daf/butler/configs/storageClasses.yaml]: Typo.

Comment (arunkannawadi) [reply python/lsst/daf/butler/configs/storageClasses.yaml]: Thanks for catching it, Tim

**Comment by arunkannawadi** [comment]
I'm not sure why `mypy` is failing, given that I did not even touch python code.

**Comment by timj** [comment]
It looks like a new release of mypy came out today that is clearly a bit cleverer and so that ignore is no longer needed. You can remove the ignore.

**Comment by arunkannawadi** [comment]
mypy workflow is failing on main itself. I could add a commit may be removing the `type: ignore`

**Comment by timj** [comment]
That's what I was trying to say. A new version of mypy came out today and so everything is going to break until we fix the code. Please fix the warning by removing the ignore.

**Review by TallJimbo — APPROVED** [review]

**Pull request merged**